{
  try {
    exchange.getUnitOfWork().done(exchange);
  }
 catch (  Throwable e) {
    LOG.warn("Exception occurred during done UnitOfWork for Exchange: " + exchange + ". This exception will be ignored.");
  }
  try {
    uow.stop();
  }
 catch (  Throwable e) {
    LOG.warn("Exception occurred during stopping UnitOfWork for Exchange: " + exchange + ". This exception will be ignored.");
  }
  exchange.setUnitOfWork(null);
}
